src/main/java/org/terasology/spawning/SpawnerSystem.java

```diff
@@ -68,232 +50,49 @@
-                        logger.info("Found a valid spawn position that can fit the Spawnable! {}", spawnPos);
-                    }
-                }
-
-                // Pick random type to spawn from the Spawner's list of types then test the cache for matching prefabs
-                String chosenSpawnerType = spawnerComp.types.get(random.nextInt(spawnerComp.types.size()));
-                Set randomType = typeLists.get(chosenSpawnerType);
-                logger.info("Picked random type {} which returned {} prefabs", chosenSpawnerType, randomType.size());
-                if (randomType.size() == 0) {
-                    logger.warn("Type {} wasn't found, sad :-( Won't spawn anything this time", chosenSpawnerType);
-                    return;
-                }
-
-                // Now actually pick one of the matching prefabs randomly and that's what we'll try to spawn
-                int anotherRandomIndex = random.nextInt(randomType.size());
-                Object[] randomPrefabs = randomType.toArray();
-                Prefab chosenPrefab = (Prefab) randomPrefabs[anotherRandomIndex];
-                logger.info("Picked index {} of types {} which is a {}, to spawn at {}", anotherRandomIndex, chosenSpawnerType, chosenPrefab, spawnPos);
-
-                // Finally create the Spawnable. Assign parentage so we can tie Spawnables to their Spawner if needed
-                EntityRef newSpawnableRef = entityManager.create(chosenPrefab, spawnPos);
-
-                logger.info("Spawning a prefab with a SKELETAL mesh: {}", chosenPrefab);
-
-                // Temp hack - make portal spawned fancy mobs bounce around like idiots too just so they do something
-                SimpleAIComponent simpleAIComponent = new SimpleAIComponent();
-                newSpawnableRef.addComponent(simpleAIComponent);
-
-                SpawnableComponent newSpawnable = newSpawnableRef.getComponent(SpawnableComponent.class);
-                newSpawnable.parent = entity;
-            }
-
-        } finally {
-            PerformanceMonitor.endActivity();
+    @ReceiveEvent
+    public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, EntityRef entity) {
+        if (event.getActionId().startsWith("Spawning-")) { //TODO Action Id can be better
+            logger.info("In Spawning");
+            EntityBuilder entityBuilder = entityManager.newBuilder(prefabMap.get(event.getActionId()));
+            SpawnableComponent spawnableComponent = entityBuilder.getComponent(SpawnableComponent.class);
+            LocationComponent locationComponent = entityBuilder.getComponent(LocationComponent.class);
+            Vector3f spawnPosition = localPlayer.getPosition(new Vector3f());
+            float v = new FastRandom().nextFloat(0, 1);
+            spawnPosition.x += v * spawnableComponent.radiusFromPlayer;
+            spawnPosition.z += Math.sqrt(1 - Math.pow(v, 2)) * spawnableComponent.radiusFromPlayer;
```

Comment:
I see, this is using the random _v_ as the _x_ component, and then calculating a _z_ component so the total works out to be exactly _radiusFromPlayer,_ right?

I think your _x_ and _z_ offsets here are always positive, which means all the spawns will end up in one quadrant instead of being distributed evenly around the circle.

Reply:
Oh yes, definitely a mistake. Fixing it.